src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -594,6 +594,14 @@ static int do_map()
   }
 
   size = info.size;
+
+  if (size > (1UL << 32) * 512) {
+    r = -EINVAL;
```

Comment:
Not sure. E2BIG is usually used when user input parameters are too big. This looks a little different. Still I don't have a strong opinion. @dillaman what do you think?